logstash-core/src/main/java/org/logstash/Event.java

```diff
@@ -264,16 +264,16 @@ public Event clone() throws CloneNotSupportedException {
     }
 
     public String toString() {
-        // TODO: (colin) clean this IOException handling, not sure why we bubble IOException here
-        try {
-            return (getTimestamp().toIso8601() + " " + this.sprintf("%{host} %{message}"));
-        } catch (IOException e) {
-            String host = (String)this.data.get("host");
-            host = (host != null ? host : "%{host}");
+        Object f = this.getField("host");
```

Comment:
For clarity I would really urge against variable reassignment. Obviously they both work, but mutating variables outside of control structures like loops usually implies that the thing behind that something is changing, which in this case it is not, we are storing two different things in the same label. It's not a pattern I see elsewhere in Logstash. This argument has been made well here: http://wiki.c2.com/?AssignVariablesOnce .  WDYT?